packages/ui-extensions/src/surfaces/customer-account/targets.ts

```diff
@@ -230,3 +230,37 @@ export interface FullPageApi {
 export interface ActionExtensionApi {
   close(): void;
 }
+
+/**
+ * For a given extension target, returns the type of the API that the
+ * extension will receive at runtime.
+ */
+export type ApiForExtension<Target extends keyof ExtensionTargets> =
+  ExtractedApiFromExtension<ExtensionTargets[Target]>;
+
+type ExtractedApiFromRunExtension<T> = T extends RunExtension<
+  infer Api,
+  unknown
+>
+  ? Api
+  : never;
+
```

Comment:
These types are from [the main branch. ](https://github.com/Shopify/ui-extensions/blob/main/packages/customer-account-ui-extensions/src/extension-points/types.ts)